src/systemctl/systemctl-edit.c

```diff
@@ -400,6 +460,42 @@ static int find_paths_to_edit(sd_bus *bus, char **names, char ***paths) {
         return 0;
 }
 
+static int trim_edit_markers(const char *path) {
+        _cleanup_free_ char *contents = NULL;
+        char *contents_start = NULL;
+        const char *contents_end = NULL;
+        size_t size;
+        int r;
+
+        /* Trim out the lines between the two markers */
+        r = read_full_file(path, &contents, &size);
+        if (r < 0) {
+                log_error_errno(r, "Failed to read temporary file \"%s\": %m", path);
+                return r;
+        }
+
+        contents_start = strstr(contents, EDIT_MARKER_START);
+        if (contents_start)
+                contents_start += strlen(EDIT_MARKER_START);
+        else
+                contents_start = contents;
+
+        contents_end = strstr(contents_start, EDIT_MARKER_END);
+        if (contents_end)
+                strshorten(contents_start, contents_end - contents_start);
+
+        contents_start = strstrip(contents_start);
+
+        /* Write new contents if the trimming actually changed anything */
+        if (strlen(contents) != size) {
+                r = write_string_file(path, contents_start, WRITE_STRING_FILE_CREATE | WRITE_STRING_FILE_TRUNCATE | WRITE_STRING_FILE_AVOID_NEWLINE);
+                if (r < 0)
+                        log_error_errno(r, "Failed to modify temporary file \"%s\": %m", path);
```

Comment:
hmm, i'd really return here directly on error, instead of catching the error below again, i.e. use:

```c
if (r < 0)
        return log_error_errno(…
```

which would then allow you to change the `return r` below to just `return 0`. I much prefer that I must say, as this means errors are handled in individual code paths that result in function exiting, and success means we reach the end of the function, but the two codepaths are never joined together again inside the function. That's a sexy property I think, if you follow what i mean